### PR TITLE
Do not set basic auth if no username/password provided

### PR DESCRIPTION
**Pull request by jdolitsky**
Not able to perform an anonymous request against Docker Hub if no username and password provided when setting up a client

### CODE REVIEW

File: opencontainers/distribution/reggie/client.py [hunk 1]
```diff
@@ -222,9 +222,11 @@ def retryRequestWithAuth(self, originalRequest, originalResponse):
             .SetQueryParam("service", h.Service)
             .SetHeader("Accept", "application/json")
             .SetHeader("User-Agent", self.Config.UserAgent)
-            .SetBasicAuth(self.Config.Username, self.Config.Password)
         )
 
+        if self.Config.Username and self.Config.Password:
```
Comment (vsoch) [inline opencontainers/distribution/reggie/client.py:227]: Could you add a comment here? I know "the code speaks for itself" but I like to be verbose so it's explicitly clear. The title of the PR would work!

File: opencontainers/version.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
 # with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-__version__ = "0.0.13"
+__version__ = "0.0.14"
```
Comment (vsoch) [inline opencontainers/version.py:7]: Perfecto! Could you add a line to the CHANGELOG to correspond with this version?

**Review by vsoch — COMMENTED** [review]
Thanks @jdolitsky ! A few tweaks below.

File: opencontainers/distribution/reggie/client.py (hunk 1 above)
Comment (jdolitsky) [reply opencontainers/distribution/reggie/client.py:228]: done!

File: opencontainers/version.py (hunk 2 above)
Comment (jdolitsky) [reply opencontainers/version.py:7]: done!

**Comment by vsoch** [comment]
Unpin the black== in the workflow - likely this was some niche upgrade error in black a while back and now the old version just doesn't work!

**Comment by vsoch** [comment]
That's also out of scope of your PR so I'm happy to do it, let me know your preference!

**Review by vsoch — APPROVED** [review]

**Comment by vsoch** [comment]
Thanks @jdolitsky !

https://pypi.org/project/opencontainers/0.0.14/

**Pull request merged**